File: DataStructures/Stack/stack.py

class Stack:

    def __init__(self, max_size):

        self.data = [0] * max_size
        self.max_size = max_size
        self.pointer = 0


    def is_full(self):

        return self.pointer == self.max_size

    def is_empty(self):

        return self.pointer == 0

    def push(self, value):

        if self.is_full(): return False

        self.data[self.pointer] = value
        self.pointer += 1

        return True

    def pop(self):

        if self.is_empty(): return (False, 0)

        self.pointer -= 1
        
        return (True, self.data[self.pointer])

    

stack = Stack(6)

for i in range(10):

    if not stack.push(i):

        print("Stack is Full!")
        break

    print("Push -> ", i)

while True:

    (success, value) = stack.pop()

    if not success:

        print("Stack is Empty!")
        break

    print("Pop -> ", value)

File: DataStructures/Stack/stack.js
class Stack {
  constructor(max_size) {
    this.data = new Array(max_size);
    this.max_size = max_size;
    this.pointer = 0;
  }

  isFull() {
    return this.pointer == this.max_size;
  }

  isEmpty() {
    return this.pointer == 0;
  }

  push(value) {
    if (this.isFull()) return false;
    this.data[this.pointer++] = value;
    return true;
  }

  pop() {
    if (this.isEmpty()) return { success: false, value: 0 };
    return { success: true, value: this.data[--this.pointer] };
  }
}

stack = new Stack(6);

for (i = 0; i < 10; i++) {
  if (!stack.push(i)) {
    console.log("Stack is Full!");
    break;
  }

  console.log("Push -> ", i);
}

while (true) {
  let { success, value } = stack.pop();

  if (!success) {
    console.log("Stack is Empty!");
    break;
  }

  console.log("Pop -> ", value);
}
